os/mm/mm_heap/mm_heapinfo_group.c

```diff
@@ -0,0 +1,189 @@
+ * mm/mm_heap/mm_heapinfo_group.c
+ *
+ *   Copyright (C) 2007, 2009, 2013-2014 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <tinyara/sched.h>
+#include <tinyara/mm/mm.h>
+#include <stdint.h>
+#include <string.h>
+#include <tinyara/mm/heapinfo_internal.h>
+
+int max_group;
+struct heapinfo_group_s heapinfo_group[HEAPINFO_USER_GROUP_NUM];
+struct heapinfo_group_info_s group_info[HEAPINFO_THREAD_NUM];
+
+/****************************************************************************
+ * Name: heapinfo_update_group
+ *
+ * Description:
+ * Update Peak heap size for Group
+ ****************************************************************************/
+void heapinfo_update_group(mmsize_t size, pid_t pid)
+{
+	int check_idx;
+	int group_num;
+	int stack_pid;
+	for (check_idx = 0; check_idx < HEAPINFO_THREAD_NUM; check_idx++) {
+		if (pid == group_info[check_idx].pid) {
+			group_num = group_info[check_idx].group;
+			heapinfo_group[group_num].curr_size += size;
+			/* Update peak size */
+			if (heapinfo_group[group_num].curr_size > heapinfo_group[group_num].peak_size) {
```

Comment:
When you change something under the branch state, below is good habit.
```
if (a > b) {
   a = something;
}
```
so I think you'd better reverse if condition.